cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -35,6 +35,13 @@ type RestorePreamble struct {
 	MinResourceFee  int64  `json:"minResourceFee,string"`
 }
 
+// LedgerEntryDiff designates a change in a ledger entry. Before and After cannot be be omitted at the same time.
+// If Before is omitted, it constitutes a creation, if After is omitted, it constitutes a delation.
+type LedgerEntryDiff struct {
```

Comment:
Has there been any discussion about also including ledger keys XDR? While all ledger keys are constructable from the ledger entry, someone has to write code to do it and it might be helpful to also export the ledger key because it's common to surface IDs in APIs alongside their data for telling the overall story for the data across different APIs. For example, the RPC function for getting ledger entries accepts a ledger key as the ID.

Reply:
Uhm, it feels a bit redundant, but it matches what we do in [getLedgerEntries](https://developers.stellar.org/network/soroban-rpc/methods/getLedgerEntries), so I will do that.